src/extension/utils/enigma-session.ts

```diff
@@ -0,0 +1,168 @@
+import { create } from 'enigma.js';
+import { buildUrl } from "enigma.js/sense-utilities";
+import schema from "enigma.js/schemas/12.20.0.json";
+import WebSocket from "ws";
+
+/**
+ * Services to create, cache and handle enigma session
+ */
+export class EnigmaSessionManager {
+
+    private static GLOBAL_SESSION_KEY = "engineData";
+
+    /**
+     * array of active session id's
+     */
+    private activeStack: Array<string>;
+
+    /**
+     * all sessions which exists
+     */
+    private sessionCache: Map<string, enigmaJS.IGeneratedAPI>;
+
+    /**
+     * max sessions we could open by default this is 5
+     */
+    private maxSessions = 5;

```

Comment:
please add public getter and setter, this is only 5 for unlicensed docker, this should be default at least 100, <0 mostly -1 should be also possible for "unlimited"

Reply:
added setter and getter for maxSessions and a value lte 0 means now infinite. By default i left the maxSessionCount to 5.